ingestion_server/ingestion_server/elasticsearch_models.py

```diff
@@ -212,20 +204,152 @@ def get_popularity(raw):
+
+        These sizes are also hardcoded in the `aspect_ratio` field in
+        openverse-api/catalog/api/serializers/image_serializers.py.
+        """
+        SMALL = 640 * 480
+        MEDIUM = 1600 * 900
+        LARGE = float("inf")
+
+    class Index:
+        name = 'image'
+
+    @staticmethod
+    def database_row_to_elasticsearch_doc(row, schema):
+        source = row[schema['source']]
+        extension = Image.get_extension(row[schema['url']])
+        categories = get_categories(extension, source)
+
+        height = row[schema['height']]
+        width = row[schema['width']]
+        aspect_ratio = Image.get_aspect_ratio(height, width)
+        size = Image.get_size(height, width)
+
+        meta = row[schema['meta_data']]
+        provider = row[schema['provider']]
+        authority_boost = Image.get_authority_boost(meta, provider)
+
+        attrs = Image.get_instance_attrs(row, schema)
+        popularity = attrs['standardized_popularity']
+
+        return Image(
+            thumbnail=row[schema['thumbnail']],
+
+            categories=categories,
+            aspect_ratio=aspect_ratio,
+            size=size,
+
+            authority_boost=authority_boost,
+            max_boost=max(popularity or 1, authority_boost or 1),
+            min_boost=min(popularity or 1, authority_boost or 1),
+            **attrs,
+        )
+
+    @staticmethod
+    def get_aspect_ratio(height, width):
+        if height is None or width is None:
+            return None
+        elif height > width:
+            aspect_ratio = Image.AspectRatios.TALL.name
+        elif height < width:
+            aspect_ratio = Image.AspectRatios.WIDE.name
         else:
+            aspect_ratio = Image.AspectRatios.SQUARE.name
+        return aspect_ratio.lower()
+
+    @staticmethod
+    def get_size(height, width):
+        if height is None or width is None:
+            return None
+        resolution = height * width
+        for size in Image.ImageSizes:
+            if resolution < size.value:
+                return size.name.lower()
+
+
+class Audio(Media):
+    """
+    Represents an audio in Elasticsearch. Note that actual mappings are defined
+    in `ingestion_server.es_mapping`.
+    """
+
+    class Durations(Enum):
+        """
+        Maximum threshold for each audio duration band
+
+        These durations are also hardcoded in the `duration` field in
+        openverse-api/catalog/api/serializers/audio_serializers.py.
+        """
+        SHORT = 4 * 60 * 1e3  # under 4 minutes
+        MEDIUM = 20 * 60 * 1e3  # 4 - 20 minutes
+        LONG = float("inf")  # longer than 20 minutes
+
+    class Index:
+        name = 'audio'
+
+    @staticmethod
+    def database_row_to_elasticsearch_doc(row, schema):
+        meta = row[schema['meta_data']]
+        provider = row[schema['provider']]
+        authority_boost = Audio.get_authority_boost(meta, provider)
+
+        attrs = Image.get_instance_attrs(row, schema)
```

Comment:
`Image` or **`Audio`**? it could be `Media` also...

Reply:
Yes, pretty sure this should me `Media.get_instance_attrs` right?

Reply:
Referencing the method from any of the three class names (`Audio`, `Image` or `Media`) works just the same:
- It's a static method so that class name we use to invoke it does not matter.
- It's defined in `Media` but inherited by both `Audio` and `Image` so calling it on Media should not be necessary.

It does make the most sense to call it from the same class as it's used in so that we can override the parent definition if we need some specific fields. Updated that in 9ef51ad. The commit dismissed your reviews unfortunately 😢.